src/dom/dom.js

```diff
@@ -1738,6 +1723,10 @@ p5.prototype.createRadio = function(...args) {
   return self;
 };
 
+// Initialize counter for unique radio group names
+p5.prototype.createRadio.counter = 0;
```

Comment:
I wonder if p5's prototype is a good place to keep this counter, although I am not much aware of the code base to provide an alternative.

Reply:
thank you for the review ,  please where is the prototype class . I will be happy to keep the counter there. I am still learning the structure of the codebase.

Reply:
I see that we are not adding anything new to the `prototype` of p5 ... we are adding to `createRadio` which is already  there, so I guess this can be OK, you can ignore my previous comment

Reply:
I'm not sure attaching to `createRadio` as a property is a good idea here. If it is just something that is kept tracked of internally, it can just be a scoped variable, ie. `let counter = 0`.

Reply:
When I use  `let counter = 0`  within the function’s scope, the counter  resets with each call to `createRadio`, and causes duplicate names like `radioOption_0` for multiple radio groups.  When I attach the counter to `p5.prototype.createRadio`, each group gets a unique identifier (e.g., `radioOption_1`, `radioOption_2`), avoiding potential conflicts between radio groups.

Reply:
The `let` statement should be outside of the function scope, it can be in the file scope as the bundler won't leak file level local variable to the larger environment.